apps/prairielearn/src/pages/instructorInstanceAdminPublishing/components/CourseInstancePublishingForm.tsx

```diff
@@ -0,0 +1,572 @@
+    if (currentValue) {
+      const currentDate = Temporal.PlainDateTime.from(currentValue);
+      const newValue = currentDate.add({ weeks: 1 });
+      setValue(field, newValue.toString());
+      // setValue doesn't seem to trigger validation so we need to trigger it manually
+      await trigger('start_date');
+      await trigger('end_date');
+    }
+  };
+
+  // Validation
+  const validateStartDate = (value: string) => {
+    if (selectedStatus === 'publish_scheduled') {
+      if (!value) {
+        return 'Start date is required for scheduled publishing';
+      }
+      // Check if start date is in the future
+      const startDateTime = plainDateTimeStringToDate(value, courseInstance.display_timezone);
+      if (startDateTime <= nowRoundedToSeconds()) {
+        return 'Start date must be in the future for scheduled publishing';
+      }
+    }
+    return true;
+  };
+
+  const validateEndDate = (value: string) => {
+    if (selectedStatus !== 'unpublished') {
+      if (!value) {
+        return 'End date is required';
+      }
+      // Check if end date is after start date
+      if (startDate && value) {
+        const startDateTime = plainDateTimeStringToDate(startDate, courseInstance.display_timezone);
+        const endDateTime = plainDateTimeStringToDate(value, courseInstance.display_timezone);
+        if (endDateTime <= startDateTime) {
+          return 'End date must be after start date';
+        }
+      }
+    }
+    return true;
+  };
+
+  return (
+    <>
+      <div class="mb-4">
+        <h4 class="mb-4">Publishing</h4>
+
+        {!canEdit && origHash !== null && (
+          <div class="alert alert-info" role="alert">
+            You do not have permission to edit access control settings.
+          </div>
+        )}
+        {!canEdit && origHash === null && (
+          <div class="alert alert-warning" role="alert">
+            You cannot edit access control settings because the <code>infoCourseInstance.json</code>{' '}
+            file does not exist.
+          </div>
+        )}
+
+        <form method="POST" onSubmit={onSubmit}>
+          <input type="hidden" name="__csrf_token" value={csrfToken} />
+          <input type="hidden" name="__action" value="update_access_control" />
+          <input type="hidden" name="orig_hash" value={origHash ?? ''} />
+
+          {/* Status Radio Buttons */}
+          <div class="mb-4">
+            {/* Unpublished */}
+            <div class="mb-3">
+              <div class="form-check">
+                <input
+                  class="form-check-input"
+                  type="radio"
+                  name="status"
+                  id="status-unpublished"
+                  value="unpublished"
+                  checked={selectedStatus === 'unpublished'}
+                  disabled={!canEdit}
+                  onChange={(e) => {
+                    const target = e.target as HTMLInputElement;
```

Comment:
As noted on another PR, these should all be using `currentTarget` instead.

Reply:
Fixed in 55c48160a88c191bdb71a899849ab38849c3267c.